core/src/main/java/com/radixdlt/environment/EventProcessor.java

```diff
@@ -67,7 +67,8 @@
 import com.radixdlt.consensus.event.CoreEvent;
 
 /**
- * Processes an event. An environment is responsible for making the call into processors.
+ * Processes an event. An environment is responsible for making the call into processors. Note that
+ * this processor can be used for both types of events, local and remote.
```

Comment:
See comment below.

Reply:
It seems that it's just `VertexRequestRateInvariant` (used in two tests) that captures a remote event (`GetVerticesRequest`) using `nodeEvents` and that requires those bindings in `DispatcherModule` / `SimulationNodeEventsModule`. What do you think about adding a separate interface for this purpose, e.g. `interface RemoteEventCapture<T: RemoteEvent> { void capture(t T);  }` and then in `DispatcherModule`:

```
  @Provides
  private RemoteEventDispatcher<NodeId, GetVerticesRequest> verticesRequestDispatcher(
      Set<RemoteEventCapture<GetVerticesRequest>> capture,
      Environment environment,
      Metrics metrics) {
    var dispatcher = environment.getRemoteDispatcher(GetVerticesRequest.class);
    return (node, request) -> {
      metrics.bft().sync().requestsSent().inc();
      dispatcher.dispatch(node, request);
      capture.forEach(e -> e.capture(request));
    };
  }
```

and in `SimulationNodeEventsModule`:

```
  @ProvidesIntoSet
  private RemoteEventCapture<GetVerticesRequest> getVerticesRequestCapture(
      @Self NodeId node, NodeEvents nodeEvents) {
    return nodeEvents.processor(node, GetVerticesRequest.class);
  }
```

?

That would allow use to have a cleaner overall design where EventDispatcher/EventProcessor are only for local events and RemoteEventDispatcher / RemoteEventProcessor are remote only.

Reply:
Check out my commit on this branch: `feature/NODE-646-follow-up-for-NODE-427-local-events`; something like this should work

Reply:
this comment still needs an update after recent changes :)